src/plugins/discover/public/components/discover_grid/discover_grid_document_selection.tsx

```diff
@@ -51,7 +57,7 @@ export const SelectButton = ({ rowIndex, setCellProps }: EuiDataGridCellValueEle
   return (
     <EuiCheckbox
       id={id}
-      label=""
+      aria-label={toggleDocumentSelectionLabel}
```

Comment:
Based on `checked` we could also show different labels: `Select ...` and `Deselect ...`

Reply:
I would like to considering having the label always say `Select ...` 

Screen readers all give a verbal confirmation of a checkbox's state by saying `<LABEL>, checkbox checked` or `<LABEL>, checkbox unchecked`. If we change the label to Deselect, our message string might say "Deselect document 1, checkbox checked" which feels like a higher mental load than "Select document 1, checkbox checked".